src/realm/sync/noinst/migration_store.hpp

```diff
@@ -0,0 +1,90 @@
+/*************************************************************************
+ *
+ * Copyright 2023 Realm, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#pragma once
+
+#include <realm/db.hpp>
+#include <realm/keys.hpp>
+#include <realm/sync/config.hpp>
+#include <realm/sync/subscriptions.hpp>
+
+#include <functional>
+#include <memory>
+#include <mutex>
+#include <string>
+
+namespace realm::sync {
+
+class MigrationStore;
+using MigrationStoreRef = std::shared_ptr<MigrationStore>;
+
+// A MigrationStore manages the PBS -> FLX migration metadata table.
+class MigrationStore : public std::enable_shared_from_this<MigrationStore> {
+public:
+    MigrationStore(const MigrationStore&) = delete;
+    MigrationStore& operator=(const MigrationStore&) = delete;
+
+    enum class MigrationState {
+        InProgress,
+        Completed,
```

Comment:
I'm not sure if we'll need 2 (NotMigrated, Complete) or 3 (NotMigrated, InProgress, Complete) states, but we can finalize this as we go along with development.

Reply:
I agree we may need more states, but it's not clear at this point.

Reply:
For now, let's just use `Migrated` and `NotMigrated` or something like that to indicate whether or not we're in the migrated state.